pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -316,6 +331,18 @@ func (c *Compactor) starting(ctx context.Context) (err error) {
 		return errors.Wrap(err, "unable to start compactor subservices")
 	}
 
+	if c.sizeBasedRetention != nil {
+		c.sizeBasedRetention.RetentionLoop = services.NewTimerService(
+			c.sizeBasedRetention.RetentionLoopInterval,
+			nil,
+			c.sizeBasedCompaction,
```

Comment:
This changes to be a function on this type so we have access to all the compaction goodness.